includes/discovery/sensors/state/cisco.inc.php

```diff
@@ -135,26 +136,31 @@
         }
 
         foreach ($temp as $index => $entry) {
-            //Discover Sensors
-            $descr = ucwords($temp[$index][$tablevalue[3]]);
-            if ($state_name == 'cRFStatusUnitState' || $state_name == 'cRFStatusPeerUnitState' || $state_name == 'cRFCfgRedundancyOperMode' || $state_name == 'cswRingRedundant') {
-                $descr = $tablevalue[3];
-            } elseif ($state_name == 'cswSwitchRole') {
-                $swrolenumber++;
-                $descr = $tablevalue[3] . $swrolenumber;
-            } elseif ($state_name == 'cswSwitchState') {
-                $swstatenumber++;
-                $descr = $tablevalue[3] . $swstatenumber;
-            } elseif ($state_name == 'cswStackPortOperStatus') {
-                $stack_port_descr = get_port_by_index_cache($device, $index);
-                $descr = $tablevalue[3] . $stack_port_descr['ifDescr'];
-            } elseif ($state_name == 'cefcFRUPowerOperStatus') {
-                $descr = snmp_get($device, 'entPhysicalName.'.$index, '-Oqv', 'ENTITY-MIB');
+            if ($tablevalue[2] == 'ciscoEnvMonTemperatureState' && (is_null($temp[$index][$tablevalue[3]]) or empty($temp[$index][$tablevalue[3]]))) {
```

Comment:
`or` should be a `||` to be consistent.

Is `empty()` not enough here or is `is_null()` needed as well?

Reply:
Yeah, just empty should be fine: http://php.net/manual/en/types.comparisons.php